File: 2k24/01 September/01-Largest Number/code.cpp
class Solution {
public:
    static bool cmp(int & a, int & b){
        string sa=to_string(a);
        string sb=to_string(b);
        return sa+sb>sb+sa;
    }
    string largestNumber(vector<int>& nums) {
        sort(nums.begin(), nums.end(), cmp);
        string res="";
        if(nums[0]==0){return "0";}
        for(auto it: nums){
            res+=to_string(it);
        }
        return res;

    }
};

File: 2k24/01 September/01-Largest Number/code.py
from typing import List
class Solution:
    @staticmethod
    def cmp(a: str, b: str)->bool::
    s1=a
    s2=b
    return (s1+s2)>(s1+s2)
def largestNumber(self, nums: List[int])-> str:
    str_num=list(map(str, nums))
    str_num.sort(key=lambda x: x*10, reverse=True)
    if(str_num[0]=="0"):
        return "0"
    return ''.join(str_num)